spharpy/spherical.py

```diff
@@ -1005,3 +1007,413 @@ def sph_identity_matrix(n_max, type='n-nm'):
+
+        self.n_max = n_max
+        self.coordinates = coordinates
+        self.basis_type = basis_type
+        self.inverse_transform = inverse_transform
+        self.channel_convention = channel_convention
+        self.normalization = normalization
+        self.phase_convention = phase_convention
+
+        self._recompute_basis = True
+        self._recompute_basis_gradient = True
+        self._recompute_inverse = True
+        self._recompute_inverse_gradient = True
+
+        self._basis = None
+        (self._basis_gradient_theta,
+         self._basis_gradient_phi) = (None, None)
+        self._basis_inv = None
+        (self._basis_inv_gradient_theta,
+         self._basis_inv_gradient_phi) = (None, None)
+
+    # Properties
+    # property for phase_convention
+    @property
+    def phase_convention(self):
+        """bool: Get phase convention."""
+        return self._phase_convention
+
+    @phase_convention.setter
+    def phase_convention(self, value):
+        """Set the phase convention."""
+        if not isinstance(value, (str, type(None))):
+            raise TypeError("phase_convention must be a string or None")
+        if value != self._phase_convention:
+            self._recompute_basis = True
+            self._recompute_basis_gradient = True
+            self._recompute_inverse = True
+            self._recompute_inverse_gradient = True
+        self._phase_convention = value
+
+    @property
+    def weights(self):
+        """array-like: Sampling weights for the quadrature transform.
+        Obtained from the coordinates object and are
+        normalized to a sum of 4*pi."""
+        if self._weights is not None:
+            return self._weights
+        elif hasattr(self.coordinates, 'weights') and self.coordinates.weights is not None:
+            return self.coordinates.weights
+        else:
+            # Calculate weights if not already set
+            self._weights = calculate_sampling_weights(self.coordinates)
+            return self._weights
+
+    @weights.setter
+    def weights(self, value):
+        self._weights = value
+
+    @property
+    def n_max(self):
+        """int: Get the spherical harmonic order."""
+        return self._n_max
+
+    @n_max.setter
+    def n_max(self, value):
+        """Set the spherical harmonic order."""
+        if value < 0:
+            raise ValueError("n_max must be a positive integer")
+        if value % 1 != 0:
+            raise ValueError("n_max must be an integer value")
+        if hasattr(self, 'channel_convention') and self.channel_convention == "fuma" and value > 3:
+            raise ValueError("n_max cannot be greater than 3 with 'fuma' channel convention")
+        if value != self._n_max:
+            self._recompute_basis = True
+            self._recompute_basis_gradient = True
+            self._recompute_inverse = True
+            self._recompute_inverse_gradient = True
+        self._n_max = int(value)  # Cast to int for safety
```

Comment:
Could be moved into the if case.
```suggestion
            self._n_max = int(value)  # Cast to int for safety
```